### PR TITLE
feat(documentation): explain custom/new postgres data types

### PR DESCRIPTION
**Pull request by javiertury**
### Pull Request check-list

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Issue #10114. I propose to document how to add custom data types. In this pull request I illustrate only how to add postgres data types.

### CODE REVIEW

**Comment by sushantdhiman** [comment]
I suggest create a new page called `Datatypes`, move stuff from https://github.com/sequelize/sequelize/blob/master/docs/models-definition.md#data-types and then add a section about customization

File: docs/data-types.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
+## Basic data types
```
Comment (sushantdhiman) [inline docs/data-types.md]: Begin with `# Datatypes`, so nav item is also named as `Datatypes`

Comment (javiertury) [reply docs/data-types.md]: Done.

**Comment by sushantdhiman** [comment]
Thanks @javiertury

**Comment by janmeier** [comment]
:+1: I've been meaning to write something like this for ages! :)

**Pull request merged**